src/components/CodeOverview/index.tsx

```diff
@@ -0,0 +1,217 @@
+
+    const availableHeight =
+      overviewHeight -
+      // Remove the top and bottom div padding.
+      overviewPadding * 2 -
+      // Adjust for the first row not having padding.
+      rowTopPadding;
+
+    const numberOfRows = Math.floor(availableHeight / rowHeight);
+
+    let chunkSize = 1;
+    if (allLineShapes.length > numberOfRows) {
+      // Split file lines evenly between all rows.
+      chunkSize = Math.ceil(allLineShapes.length / numberOfRows);
+    }
+
+    return {
+      numberOfRows,
+      chunkedLineShapes: chunk(allLineShapes, chunkSize),
+    };
+  }
+
+  renderRow(
+    selectedMessageMap: LinterProviderInfo['selectedMessageMap'],
+    rowIndex: number,
+    shapeIndex: number,
+    groupOflineShapes: LineShapes[] | void,
+  ) {
+    if (!groupOflineShapes) {
+      return null;
+    }
+
+    // TODO: render linter messages here by looking at all shapes
+    // in groupOflineShapes.
+    // See https://github.com/mozilla/addons-code-manager/issues/523
+
+    const lineShapes = groupOflineShapes[shapeIndex];
+
+    return <CodeLineShapes lineShapes={lineShapes} />;
+  }
+
+  renderOverview(selectedMessageMap: LinterProviderInfo['selectedMessageMap']) {
+    const { content, location, rowHeight, rowTopPadding } = this.props;
+    const { overviewHeight } = this.state;
+
+    if (!overviewHeight) {
+      return null;
+    }
+
+    const allLineShapes = generateLineShapes(getLines(content));
+    const { numberOfRows, chunkedLineShapes } = this.fitLineShapesIntoOverview(
+      allLineShapes,
+    );
+    const overview = [];
+
+    for (let rowIndex = 0; rowIndex < numberOfRows; rowIndex++) {
+      const shapes = chunkedLineShapes[rowIndex] || undefined;
+      // Use the first line in the group.
+      const shapeIndex = 0;
+      const line =
+        shapes && shapes.length ? shapes[shapeIndex].line : undefined;
+
+      overview.push(
+        <Link
+          className={styles.line}
+          to={{
+            ...location,
+            hash: line ? getCodeLineAnchor(line) : '#',
+          }}
+          key={rowIndex}
+          style={{
+            height: `${rowHeight}px`,
+            paddingTop: rowIndex > 0 ? `${rowTopPadding}px` : undefined,
+          }}
+          title={line ? gettext(`Jump to line ${line}`) : ''}
+        >
+          {this.renderRow(selectedMessageMap, rowIndex, shapeIndex, shapes)}
+        </Link>,
+      );
+    }
+
+    return overview;
+  }
+
+  renderWithLinterInfo = ({ selectedMessageMap }: LinterProviderInfo) => {
+    return (
+      <div
+        ref={this.overviewRef}
+        className={styles.CodeOverview}
+        style={{
+          padding: `${this.props.overviewPadding}px`,
+        }}
+      >
+        {this.renderOverview(selectedMessageMap)}
+      </div>
+    );
+  };
+
+  render() {
+    const { overviewHeight } = this.state;
+    const { version } = this.props;
+
+    return (
+      <LinterProvider
+        key={overviewHeight ? String(overviewHeight) : ''}
```

Comment:
Just wondering, why did you decide to use `overviewHeight` as the `key`?

Reply:
The key is needed because `<LinterProvider>` should re-render any time `overviewHeight` changes. Key values in React can be used to force re-renders -- this is a bit odd to see when you're unfamiliar with that magic.